src/main/java/frc/trigon/robot/subsystems/roller/Roller.java

```diff
@@ -19,100 +15,48 @@ public static Roller getInstance() {
-                (interrupted) -> stopAngleMotor(),
-                this::isClosed
-        );
-    }
-
-    /**
-     * @return a command that activates the collection motor with collection power
-     */
-    public Command getCollectCommand() {
-        return new StartEndCommand(
-                this::collect,
-                this::stopCollection,
-                this
-        );
+    @Override
+    public void periodic() {
+        rollerIO.updateInputs(rollerInputs);
+        Logger.processInputs("Roller", rollerInputs);
+        updateMechanism();
     }
 
-    /**
-     * @return a command that stops the collection motor
-     */
-    public Command getStopCollectionCommand() {
-        return new InstantCommand(
-                this::stopCollection,
-                this
-        );
+    void openRoller() {
+        rollerIO.setAngleMotorPower(RollerConstants.OPEN_POWER);
     }
 
-    /**
-     * @return a command that opens the angle of the roller and activates the collection motor
-     */
-    public Command getFullOpeningCommand() {
-        return new ParallelCommandGroup(
-                getOpenRollerCommand(),
-                Commands.removeRequirements(getCollectCommand())
-        );
+    void closeRoller() {
+        rollerIO.setAngleMotorPower(RollerConstants.CLOSE_POWER);
     }
 
-    /**
-     * @return a command that closes the angle of the roller and stops the collection motor
-     */
-    public Command getFullStopCommand() {
-        return new ParallelCommandGroup(
-                getCloseRollerCommand(),
-                Commands.removeRequirements(getStopCollectionCommand())
-        );
+    void collect() {
+        rollerIO.setCollectionMotorPower(RollerConstants.COLLECTION_MOTOR_SPEED);
     }
 
-    private void openRoller() {
-        angleMotor.set(ControlMode.PercentOutput, RollerConstants.OPEN_POWER);
+    boolean isOpen() {
+        return !rollerInputs.forwardLimitSwitchPressed;
     }
 
-    private void closeRoller() {
-        angleMotor.set(ControlMode.PercentOutput, RollerConstants.CLOSE_POWER);
+    boolean isClosed() {
+        return !rollerInputs.backwardLimitSwitchPressed;
     }
 
-    private void collect() {
-        collectionMotor.set(RollerConstants.COLLECTION_MOTOR_SPEED);
+    void stopAngleMotor() {
+        rollerIO.stopAngleMotor();
     }
 
-    private void stopCollection() {
-        collectionMotor.stopMotor();
+    void stopCollectionMotor() {
+        rollerIO.stopCollectionMotor();
     }
 
-    private void stopAngleMotor() {
-        angleMotor.set(ControlMode.Disabled, 0);
+    private void updateMechanism() {
+        if (isOpen())
+            RollerConstants.ROLLER_ANGLE_LIGAMENT.setAngle(0);
+        else if (isClosed())
+            RollerConstants.ROLLER_ANGLE_LIGAMENT.setAngle(90);
+        else
+            RollerConstants.ROLLER_ANGLE_LIGAMENT.setAngle(45);
+        Logger.recordOutput("RollerMechanism", RollerConstants.ROLLER_ANGLE_MECHANISM);
```

Comment:
Log this using the `@AutoLogOutput` annotation and log it under Roller/RollerMechanism.